src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/ModelBinderFactory.cs

```diff
@@ -46,12 +46,20 @@ public IModelBinder CreateBinder(ModelBinderFactoryContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
+            if (_providers.Length == 0)
+            {
+                throw new InvalidOperationException(Resources.FormatModelBinderProvidersAreRequired(
+                    typeof(MvcOptions).FullName,
+                    nameof(MvcOptions.ModelBinderProviders),
+                    typeof(IModelBinderProvider).FullName));
+            }
```

Comment:
Why is this necessary? without this change we still report a good error message right?